src/Aspire.Dashboard/Otlp/Storage/Subscription.cs

```diff
@@ -6,20 +6,45 @@ namespace Aspire.Dashboard.Otlp.Storage;
 public sealed class Subscription : IDisposable
 {
     private readonly Func<Task> _callback;
+    private readonly ExecutionContext? _executionContext;
     private readonly Action _unsubscribe;
 
     public string? ApplicationId { get; }
     public SubscriptionType SubscriptionType { get; }
 
-    public Subscription(string? applicationId, SubscriptionType subscriptionType, Func<Task> callback, Action unsubscribe)
+    public Subscription(string? applicationId, SubscriptionType subscriptionType, Func<Task> callback, Action unsubscribe, ExecutionContext? executionContext)
     {
         ApplicationId = applicationId;
         SubscriptionType = subscriptionType;
         _callback = callback;
         _unsubscribe = unsubscribe;
+        _executionContext = executionContext;
     }
 
-    public Task ExecuteAsync() => _callback();
+    public async Task ExecuteAsync()
+    {
+        // Set the execution context to the one captured when the subscription was created.
+        // This ensures that the callback runs in the same context as the subscription was created.
+        // For example, the request culture is used to format content in the callback.
+
+        var current = ExecutionContext.Capture();
```

Comment:
why are we capturing again?

Reply:
Because https://github.com/dotnet/aspire/issues/2554

There is a comment immediately above the capture. What should it say to be clearer?